distribution/tools/keystore-cli/src/main/java/org/elasticsearch/common/settings/cli/ShowKeyStoreCommand.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.settings.cli;
+
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.settings.BaseKeyStoreCommand;
+import org.elasticsearch.common.settings.KeyStoreWrapper;
+import org.elasticsearch.env.Environment;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.CodingErrorAction;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * A subcommand for the keystore cli to show the value of a setting in the keystore.
+ */
+public class ShowKeyStoreCommand extends BaseKeyStoreCommand {
+
+    private final OptionSpec<String> arguments;
+
+    public ShowKeyStoreCommand() {
+        super("Show a value from the keystore", true);
+        this.arguments = parser.nonOptions("setting name");
+    }
+
+    @Override
+    protected void executeCommand(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        final List<String> names = arguments.values(options);
+        if (names.size() != 1) {
+            throw new UserException(ExitCodes.USAGE, "Must provide a single setting name to show");
+        }
+        final String settingName = names.get(0);
+
+        final KeyStoreWrapper keyStore = getKeyStore();
+        if (keyStore.getSettingNames().contains(settingName) == false) {
+            throw new UserException(ExitCodes.CONFIG, "Setting [" + settingName + "] does not exist in the keystore.");
+        }
+
+        try (InputStream input = keyStore.getFile(settingName)) {
+            final BytesReference bytes = org.elasticsearch.common.io.Streams.readFully(input);
+            final OutputStream output = terminal.getOutputStream();
+            if (output != null) {
+                bytes.writeTo(output);
```

Comment:
This assumes that the terminal is using UTF8, which may not be the case. I think we should flip this around, so that we first try decoding utf8, and print if possible, then fall back to writing to the output stream (when we know the data is not utf8).

Reply:
The reason I had it in this sequence is because in my (somewhat limited) testing, we never seem to have an output stream unless writing to a file, and if we're writing to a file then we probably want byte-equivalent output.

If we flip it around then
```
bin/elasticsearch-keystore add-file some.setting my-file1
bin/elasticsearch-keystore show some.setting > my-file2
``` 
might mean that `my-file1` and `my-file2` are not byte identical. At the very least we need to decide whether to append a newline or not (ideally with file output we wouldn't, with terminal output we would).

I'll make the switch in the sequence and select `print` vs `println` depending on whether the terminal has an output stream. It's not ideal, but I don't think there's a perfect answer.